src/governance.ts

```diff
@@ -418,23 +418,27 @@ export class GovernanceCanister {
   /**
    * Gets the NeuronID of a newly created neuron.
    */
-  public claimOrRefreshNeuronFromAccount = async (
-    request: ClaimOrRefreshNeuronFromAccount
-  ): Promise<NeuronId | undefined> => {
-    // Note: This is an update call so the certified and uncertified services are identical in this case,
-    // however using the certified service provides protection in case that changes.
-    const response =
-      await this.certifiedService.claim_or_refresh_neuron_from_account({
-        controller: request.controller ? [request.controller] : [],
-        memo: request.memo,
-      });
-
-    const { result } = response;
-    if (result.length && "NeuronId" in result[0]) {
-      return result[0].NeuronId.id;
+  public claimOrRefreshNeuronFromAccount = async ({
+    memo,
+    controller,
+  }: {
+    memo: bigint;
+    controller: Principal;
+  }): Promise<NeuronId | undefined> => {
+    const rawRequest = toClaimOrRefreshRequest({
+      memo,
+      controller,
+    });
+    const rawResponse = await this.certifiedService.manage_neuron(rawRequest);
+    const { command } = rawResponse;
+    if (command.length && "ClaimOrRefresh" in command[0]) {
```

Comment:
couldn't / shouldn't you use `toCommand` from `response.converter.ts` that contains already a `"ClaimOrRefresh"`?

Reply:
Do you mean inside the `toClaimOrRefreshRequest`?

`toCommand` needs a neuronId:

```
export const toCommand = ({
  neuronId,
  command,
}: {
  neuronId: NeuronId;
  command: RawCommand;
}): RawManageNeuron => ({
  id: [{ id: neuronId }],
  command: [command],
  neuron_id_or_subaccount: [],
});
```

I would have to make it optional and add the functionality there just for one use case. That's why I thought it made more sense to have this use case separately.

Reply:
no. i mean following already exists: https://github.com/dfinity/nns-js/blob/8225630d32c6b5d5685fadf2ea85730ba0ddc6ec/src/canisters/governance/response.converters.ts#L351

it looks related to what you are adding here.

Reply:
Here I am reading the response. If I use `toCommand`, I still need to check the return value.

`toCommand` is used in `toAction`, which is used in `toProposal`. But for ManageNeuron requests, we used the helper `manageNeuron` for most configurations. The helper is used when we want to return `void`. But in the case, we want to read the response and return the neuron id from it.

Reply:
ah good. I guess later on it might be interesting to rename that `toCommand` to make its goal more obvious then. thanks for the feedback.